FiveRockingFingers/FRF.Core/Response/ErrorCodes.cs

```
@@ -11,5 +11,8 @@ public static class ErrorCodes
         // Relation errors
         public const int RelationAlreadyExisted = 10;
         public const int RelationNotValid = 11;
+
+        // Users
+        public const int SignUpError = 22;
```

Remember that everything listed here is an error. What does this really means?

---

Here you can have something like "_EmailAlreadyExist_" and "_AWSCognitoServerError_".

---

This can mean multiple things. I'm not quite sure what to call it, and I'm not sure about how specific it should be. As far as I know, letting someone know that an email already exists in the database can be considered a security flaw, since it lets people know there's an account registered with that email, and then paired with brute force could let someone into the system. That's the reasoning behind this name, but this are all internal errors, business ones I think we called them, so I don't know how specific they should be.

---

Good chat, I like it. Keep them coming.

Indeed exposing something like "email already exist" needs to be prevented if possible.

"Invalid credentials" is generic enough without telling if the username (or email) or password was wrong.

Regarding AWS Cognito error is telling about internal implementation details -- which we should avoid as well. Do we know why AWS Cognito can fail? Is this the error when the credentials are wrong? Or this means the AWS Cognito server is down, or something else?

---

I think it would fail when AWS Cognito servers are down, but I'm not sure. Maybe @gonzalorom knows more.

---

Like Matias said "Invalid credentials" is generic enough, because even if we have a password rule in the model binding this can cause a not succeeded response. With the AWS Cognito error, most of the time when the server is down, or you have an invalid secret_pool_id or just right after the creation of the user is done, you change the scope of your IAM AWS user, you will get this error. All these scenarios, in my opinion, are a bit rare and the end user can't do anything about it.

---

Ok so as far as I understand then we need two errors: Invalid credentials and one for the AWS cognito error which should never happen. Maybe we can call this one AuthenticationServerCurrentlyUnavailable or something along those lines? And (I don't like it) but we should probably return a 500 error to consumers in that case.

401 for the invalid credentials, of course.

---

Done!